File: C++/Remove all duplicates from a given string.cpp
class Solution{
public:
    string removeDuplicates(string str) {
        string res="";
        int n= str.length();
       unordered_set<char>st;
        
        for(int i=0; i< n; i++){
            char ch=str[i];
           if(st.find(ch)!=st.end()) 
                continue;
           else
           {
                st.insert(ch);
                res+=ch;
           }
        }
        return res;
    }
};

File: JAVA/Remove all duplicates from a given string.java
class Solution {
    String removeDuplicates(String str) {

      int small[]=new int[26];
      int big[]=new int[26];
      String out="";
      
      for(int i=0;i<str.length();i++)
      {
          char ch=str.charAt(i);
          if(ch>=65 && ch<=90)
          big[ch-'A']++;
          else
          small[ch-'a']++;
      }
      
      for(int i=0;i<str.length();i++)
      {
          char ch=str.charAt(i);
          if(ch>=65 && ch<=90){
          if(big[ch-'A']>0)
          out+=ch;
          
          big[ch-'A']=0;
          }
          else{
           if(small[ch-'a']>0)
          out+=ch;
          
          small[ch-'a']=0;
          }
      }
      
      return out;

    }
}
